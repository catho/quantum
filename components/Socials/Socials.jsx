import React from 'react';
import Proptypes from 'prop-types';
import styled from 'styled-components';
import SocialType from './sub-components/SocialType';
import { spacing } from '../shared/theme';

const List = styled.ul`
  padding: 0;
  display: inline-flex;
`;

const ListItem = styled.li`
  display: inline-flex;
  list-style: none;
  ${({
    theme: {
      spacing: { medium },
    },
  }) => `margin-right: ${medium}px;`}

  &:last-child {
    margin-right: 0;
  }
`;

const SocialWrapper = styled.section`
  display: inline-flex;
`;

ListItem.displayName = 'ListItem';

const Socials = ({ items, size, theme, withBox }) => (
  <SocialWrapper>
    <List>
      {items.map(item => (
        <ListItem theme={theme} key={`social-item-${item.name}`}>
          <SocialType
            key={`social-${item.name}`}
            url={item.url}
            type={item.name}
            size={size}
            title={item.title}
            withBox={withBox}
          />
        </ListItem>
      ))}
    </List>
  </SocialWrapper>
);

Socials.defaultProps = {
  size: '48',
  theme: { spacing },
  withBox: true,
};

Socials.propTypes = {
  size: Proptypes.string,
  theme: Proptypes.shape({
    spacing: Proptypes.object,
  }),
  items: Proptypes.arrayOf(
    Proptypes.shape({
      name: Proptypes.oneOf(['facebook', 'twitter', 'youtube']),
      url: Proptypes.string,
      title: Proptypes.string,
    }).isRequired,
  ).isRequired,
  withBox: Proptypes.bool,
};

export default Socials;
