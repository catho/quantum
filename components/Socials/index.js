import Socials from './Socials';

export default Socials;
