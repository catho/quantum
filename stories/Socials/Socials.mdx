import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';

import Socials from '../../components/Socials';

import { Header } from '../shared';

import { 
  Default,
  WithoutBox,
  WithSize,
  AllItems
} from './Socials.stories.jsx';
import EvaluationForm from '../../stories/shared/EvaluationForm'


<Meta title="Socials" component={Socials} />

<Header title="Socials">
  Socials are links to their respective social media platform.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import Socials from '@catho/quantum/Socials';
```

<Canvas>
  <Story of={Default} />
</Canvas>

## <a id="api"></a>API

<ArgsTable of={Socials} />

## <a id="examples"></a>Examples

### Without Box

<Canvas>
  <Story of={WithoutBox} />
</Canvas>

### All Items

<Canvas>
  <Story of={AllItems} />
</Canvas>

### With Size

<Canvas>
  <Story of={WithSize} />
</Canvas>


<EvaluationForm componentName="Socials" />




