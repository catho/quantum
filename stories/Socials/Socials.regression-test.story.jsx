import React from 'react';
import { storiesOf } from '@storybook/react';

import Socials from '../../components/Socials';

const allItems = [
  {
    name: 'facebook',
    url: 'http://facebook.com',
    description: 'facebook description of the purpose of this icon',
  },
  {
    name: 'twitter',
    url: 'http://twitter.com',
    description: 'twitter description of the purpose of this icon',
  },
  {
    name: 'youtube',
    url: 'http://youtube.com',
    description: 'youtube description of the purpose of this icon',
  },
];

storiesOf('Socials', module)
  .add('All items', () => (
    <div>
      <Socials items={allItems} />
    </div>
  ))

  .add('all items (24px)', () => (
    <div>
      <Socials items={allItems} size="24" />
    </div>
  ))
  .add('only Facebook', () => (
    <div>
      <Socials items={[allItems[0]]} />
    </div>
  ))
  .add('only Twitter', () => (
    <div>
      <Socials items={[allItems[1]]} />
    </div>
  ))
  .add('only Youtube', () => (
    <div>
      <Socials items={[allItems[2]]} />
    </div>
  ))
  .add('all items without box (24px)', () => (
    <div>
      <Socials items={allItems} withBox={false} size="24" />
    </div>
  ))
  .add('all items without box', () => (
    <div>
      <Socials items={allItems} withBox={false} />
    </div>
  ));
