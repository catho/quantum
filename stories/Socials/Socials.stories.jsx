import Socials from '../../components/Socials';

export default {
  title: 'Socials',
  component: Socials,
};

const allItems = [
  {
    name: 'facebook',
    url: 'https://facebook.com',
    title: 'facebook',
  },
  {
    name: 'twitter',
    url: 'https://twitter.com',
    title: 'twitter',
  },
  {
    name: 'youtube',
    url: 'https://youtube.com',
    title: 'youtube',
  },
];

const Template = args => <Socials items={[allItems[1]]} {...args} />;

export const Default = Template.bind({});

export const WithoutBox = Template.bind({});

WithoutBox.args = {
  withBox: false,
};

export const AllItems = Template.bind({});
AllItems.args = {
  items: allItems,
};

export const WithSize = Template.bind({});
WithSize.args = {
  size: '24',
};
