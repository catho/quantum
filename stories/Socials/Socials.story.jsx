import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  TabbedView,
  Tab,
  AutoPropsApi,
  Heading,
  StoryContainer,
  Title,
  SimpleHighlight,
} from '@catho/quantum-storybook-ui';

import { Row, Col } from '../../components';

import Socials from '../../components/Socials';
import {
  SocialExampleAll,
  SocialExampleChangingSize,
  SocialExampleFacebook,
  SocialExampleWithoutBox,
  SocialExampleWithoutBoxChangingSize,
} from './examples/SocialExamples';

storiesOf('Socials', module).add('Socials', () => (
  <>
    <Heading name="Socials">
      Socials component is used to share the social media in applications.
    </Heading>
    <TabbedView>
      <Tab title="Usage">
        <StoryContainer>
          <Title as="h2">Importing Socials</Title>
          <SimpleHighlight>{`import { Socials } from '@catho/quantum';`}</SimpleHighlight>
          <Row>
            <Col style={{ marginTop: '20px' }} />
          </Row>
        </StoryContainer>
      </Tab>

      <Tab title="API">
        <AutoPropsApi component={Socials} />
      </Tab>

      <Tab title="Example">
        <StoryContainer>
          <Title as="h1">Socials</Title>
          <p>
            Here you can check a simple implamentation using Socials component.
          </p>
          <Title as="h3">All socials</Title>
          <Row>
            <Col xsmall={4} small={4} medium={6}>
              <SimpleHighlight>{SocialExampleAll.code}</SimpleHighlight>
            </Col>
            <Col xsmall={4} small={4} medium={6}>
              <SocialExampleAll />
            </Col>
          </Row>
          <Title as="h3">All socials (24px)</Title>
          <Row>
            <Col xsmall={4} small={4} medium={6}>
              <SimpleHighlight>
                {SocialExampleChangingSize.code}
              </SimpleHighlight>
            </Col>
            <Col xsmall={4} small={4} medium={6}>
              <SocialExampleChangingSize />
            </Col>
          </Row>
          <Title as="h3">With only facebook</Title>
          <Row>
            <Col xsmall={4} small={4} medium={6}>
              <SimpleHighlight>{SocialExampleFacebook.code}</SimpleHighlight>
            </Col>
            <Col xsmall={4} small={4} medium={6}>
              <SocialExampleFacebook />
            </Col>
          </Row>
          <Title as="h3">Socials without boxes</Title>
          <Row>
            <Col xsmall={4} small={4} medium={6}>
              <SimpleHighlight>{SocialExampleWithoutBox.code}</SimpleHighlight>
            </Col>
            <Col xsmall={4} small={4} medium={6}>
              <SocialExampleWithoutBox />
            </Col>
          </Row>
          <Title as="h3">Socials without boxes (24px)</Title>
          <Row>
            <Col xsmall={4} small={4} medium={6}>
              <SimpleHighlight>
                {SocialExampleWithoutBoxChangingSize.code}
              </SimpleHighlight>
            </Col>
            <Col xsmall={4} small={4} medium={6}>
              <SocialExampleWithoutBoxChangingSize />
            </Col>
          </Row>
        </StoryContainer>
      </Tab>
    </TabbedView>
  </>
));
