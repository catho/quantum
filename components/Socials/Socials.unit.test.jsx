import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Socials from './Socials';

const allSocialsList = [
  {
    name: 'facebook',
    url: 'https://facebook.com',
    title: 'facebook title of the purpose of this icon',
  },
  {
    name: 'twitter',
    url: 'https://twitter.com',
    title: 'twitter title of the purpose of this icon',
  },
  {
    name: 'youtube',
    url: 'https://youtube.com',
    title: 'youtube title of the purpose of this icon',
  },
];

describe('<Socials />', () => {
  it('should match snapshots', () => {
    const SOCIALS = [
      <Socials items={allSocialsList} />,
      <Socials items={[allSocialsList[0]]} />,
      <Socials items={allSocialsList} size="24" />,
    ];

    SOCIALS.forEach(social => {
      const SocialComponent = mount(social);
      expect(toJson(SocialComponent)).toMatchSnapshot();
      SocialComponent.unmount();
    });
  });

  it('should pass title to <Title /> on SVG ', () => {
    const component = mount(<Socials items={[allSocialsList[0]]} />);
    expect(component.find('title').text()).toBe(allSocialsList[0].title);
  });

  it('should have the same number of social types passed in object items ', () => {
    const wrapper = mount(<Socials items={allSocialsList} />);
    expect(wrapper.find('ListItem').length).toBe(allSocialsList.length);
  });
});
