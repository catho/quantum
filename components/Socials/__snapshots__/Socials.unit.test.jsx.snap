// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Socials /> should match snapshots 1`] = `
.c5 {
  display: inline-block;
}

.c4 {
  display: inline-block;
}

.c6 {
  display: inline-block;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  padding: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  list-style: none;
  margin-right: 16px;
}

.c2:last-child {
  margin-right: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<Socials
  items={
    Array [
      Object {
        "name": "facebook",
        "title": "facebook title of the purpose of this icon",
        "url": "https://facebook.com",
      },
      Object {
        "name": "twitter",
        "title": "twitter title of the purpose of this icon",
        "url": "https://twitter.com",
      },
      Object {
        "name": "youtube",
        "title": "youtube title of the purpose of this icon",
        "url": "https://youtube.com",
      },
    ]
  }
  size="48"
  theme={
    Object {
      "spacing": Object {
        "large": 24,
        "medium": 16,
        "small": 12,
        "xlarge": 32,
        "xsmall": 8,
        "xxlarge": 40,
        "xxsmall": 4,
        "xxxlarge": 48,
        "xxxsmall": 2,
      },
    }
  }
  withBox={true}
>
  <Socials__SocialWrapper>
    <section
      className="c0"
    >
      <Socials__List>
        <ul
          className="c1"
        >
          <ListItem
            key="social-item-facebook"
            theme={
              Object {
                "spacing": Object {
                  "large": 24,
                  "medium": 16,
                  "small": 12,
                  "xlarge": 32,
                  "xsmall": 8,
                  "xxlarge": 40,
                  "xxsmall": 4,
                  "xxxlarge": 48,
                  "xxxsmall": 2,
                },
              }
            }
          >
            <li
              className="c2"
            >
              <SocialType
                key="social-facebook"
                size="48"
                title="facebook title of the purpose of this icon"
                type="facebook"
                url="https://facebook.com"
                withBox={true}
              >
                <SocialType__SocialWrapper
                  href="https://facebook.com"
                >
                  <a
                    className="c3"
                    href="https://facebook.com"
                  >
                    <SocialFacebookIcon
                      size="48"
                      title="facebook title of the purpose of this icon"
                      withBox={true}
                    >
                      <SocialFacebookIcon__StyledSvg
                        aria-labelledby="facebook-title"
                        height="48"
                        role="img"
                        viewBox="0 0 48 48"
                        width="48"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-labelledby="facebook-title"
                          className="c4"
                          height="48"
                          role="img"
                          viewBox="0 0 48 48"
                          width="48"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="facebook-title"
                          >
                            facebook title of the purpose of this icon
                          </title>
                          <g
                            id="Box_Facebook"
                            transform="translate(606 -35)"
                          >
                            <rect
                              data-name="Rectangle 1"
                              fill="#3c5193"
                              height={48}
                              id="Rectangle_1"
                              rx="9"
                              transform="translate(-606 35)"
                              width={48}
                            />
                            <path
                              d="M146.667,100.389h2.285V98.246a5.551,5.551,0,0,1,.737-3.306,4.115,4.115,0,0,1,3.547-1.606,14.874,14.874,0,0,1,4.1.4L156.763,97a7.945,7.945,0,0,0-1.842-.267c-.888,0-1.685.309-1.685,1.165v2.493h3.645l-.254,3.192h-3.391v11.086h-4.284V103.581h-2.285Z"
                              data-name="Path 4"
                              fill="#fff"
                              fillRule="evenodd"
                              id="Path_4"
                              transform="translate(-734 -45)"
                            />
                          </g>
                        </svg>
                      </SocialFacebookIcon__StyledSvg>
                    </SocialFacebookIcon>
                  </a>
                </SocialType__SocialWrapper>
              </SocialType>
            </li>
          </ListItem>
          <ListItem
            key="social-item-twitter"
            theme={
              Object {
                "spacing": Object {
                  "large": 24,
                  "medium": 16,
                  "small": 12,
                  "xlarge": 32,
                  "xsmall": 8,
                  "xxlarge": 40,
                  "xxsmall": 4,
                  "xxxlarge": 48,
                  "xxxsmall": 2,
                },
              }
            }
          >
            <li
              className="c2"
            >
              <SocialType
                key="social-twitter"
                size="48"
                title="twitter title of the purpose of this icon"
                type="twitter"
                url="https://twitter.com"
                withBox={true}
              >
                <SocialType__SocialWrapper
                  href="https://twitter.com"
                >
                  <a
                    className="c3"
                    href="https://twitter.com"
                  >
                    <SocialTwitterIcon
                      size="48"
                      title="twitter title of the purpose of this icon"
                      withBox={true}
                    >
                      <SocialTwitterIcon__StyledSvg
                        aria-labelledby="twitter-title"
                        height="48"
                        role="img"
                        viewBox="0 0 48 48"
                        width="48"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-labelledby="twitter-title"
                          className="c5"
                          height="48"
                          role="img"
                          viewBox="0 0 48 48"
                          width="48"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="twitter-title"
                          >
                            twitter title of the purpose of this icon
                          </title>
                          <g
                            id="box_twitter"
                            transform="translate(606 -203)"
                          >
                            <rect
                              data-name="twitter-name"
                              fill="#2caae1"
                              height={48}
                              id="twitterID"
                              rx="9"
                              transform="translate(-606 203)"
                              width={48}
                            />
                            <path
                              d="M166.667,262.829a11.931,11.931,0,0,1-3.457.955,6.067,6.067,0,0,0,2.64-3.355,11.977,11.977,0,0,1-3.821,1.478,6.007,6.007,0,0,0-9.872,1.655,6.107,6.107,0,0,0-.383,3.876,17.046,17.046,0,0,1-12.4-6.347,6.1,6.1,0,0,0,1.862,8.092,5.957,5.957,0,0,1-2.731-.755v.077a6.089,6.089,0,0,0,1.367,3.843,6.01,6.01,0,0,0,3.474,2.1,6.017,6.017,0,0,1-2.718.1,6.068,6.068,0,0,0,2.141,3.013,6,6,0,0,0,3.481,1.2,12.071,12.071,0,0,1-8.917,2.51A16.943,16.943,0,0,0,146.559,284c11.07,0,17.123-9.239,17.123-17.252,0-.263,0-.525-.016-.785a12.288,12.288,0,0,0,3-3.134Z"
                              data-name="Path 27"
                              fill="#fff"
                              id="twitterID-path"
                              transform="translate(-734 -45)"
                            />
                          </g>
                        </svg>
                      </SocialTwitterIcon__StyledSvg>
                    </SocialTwitterIcon>
                  </a>
                </SocialType__SocialWrapper>
              </SocialType>
            </li>
          </ListItem>
          <ListItem
            key="social-item-youtube"
            theme={
              Object {
                "spacing": Object {
                  "large": 24,
                  "medium": 16,
                  "small": 12,
                  "xlarge": 32,
                  "xsmall": 8,
                  "xxlarge": 40,
                  "xxsmall": 4,
                  "xxxlarge": 48,
                  "xxxsmall": 2,
                },
              }
            }
          >
            <li
              className="c2"
            >
              <SocialType
                key="social-youtube"
                size="48"
                title="youtube title of the purpose of this icon"
                type="youtube"
                url="https://youtube.com"
                withBox={true}
              >
                <SocialType__SocialWrapper
                  href="https://youtube.com"
                >
                  <a
                    className="c3"
                    href="https://youtube.com"
                  >
                    <SocialYoutubeIcon
                      size="48"
                      title="youtube title of the purpose of this icon"
                      withBox={true}
                    >
                      <SocialYoutubeIcon__StyledSvg
                        aria-labelledby="youtube-title"
                        height="48"
                        role="img"
                        viewBox="0 0 50 50"
                        width="48"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-labelledby="youtube-title"
                          className="c6"
                          height="48"
                          role="img"
                          viewBox="0 0 50 50"
                          width="48"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="youtube-title"
                          >
                            youtube title of the purpose of this icon
                          </title>
                          <g
                            id="Box_Youtube"
                            transform="translate(350 -35)"
                          >
                            <rect
                              data-name="Rectangle 3"
                              fill="#fff"
                              height={48}
                              id="Rectangle_3"
                              rx="8"
                              stroke="#424242"
                              strokeWidth="2"
                              transform="translate(-349 36)"
                              width={48}
                            />
                            <path
                              d="M422.05,96.9a3.711,3.711,0,0,0-2.6-2.617c-2.283-.617-11.45-.617-11.45-.617s-9.167,0-11.467.617a3.685,3.685,0,0,0-2.6,2.617,39.409,39.409,0,0,0-.6,7.1,38.348,38.348,0,0,0,.617,7.1,3.711,3.711,0,0,0,2.6,2.617c2.283.616,11.45.616,11.45.616s9.167,0,11.467-.616a3.687,3.687,0,0,0,2.6-2.617,39.409,39.409,0,0,0,.6-7.1A38.348,38.348,0,0,0,422.05,96.9Z"
                              data-name="Path 20"
                              fill="#ee3124"
                              id="Path_20"
                              transform="translate(-733 -44)"
                            />
                            <path
                              d="M405,108.367V99.633L412.667,104Z"
                              data-name="Path 21"
                              fill="#fff"
                              id="Path_21"
                              transform="translate(-733 -44)"
                            />
                          </g>
                        </svg>
                      </SocialYoutubeIcon__StyledSvg>
                    </SocialYoutubeIcon>
                  </a>
                </SocialType__SocialWrapper>
              </SocialType>
            </li>
          </ListItem>
        </ul>
      </Socials__List>
    </section>
  </Socials__SocialWrapper>
</Socials>
`;

exports[`<Socials /> should match snapshots 2`] = `
.c4 {
  display: inline-block;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  padding: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  list-style: none;
  margin-right: 16px;
}

.c2:last-child {
  margin-right: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<Socials
  items={
    Array [
      Object {
        "name": "facebook",
        "title": "facebook title of the purpose of this icon",
        "url": "https://facebook.com",
      },
    ]
  }
  size="48"
  theme={
    Object {
      "spacing": Object {
        "large": 24,
        "medium": 16,
        "small": 12,
        "xlarge": 32,
        "xsmall": 8,
        "xxlarge": 40,
        "xxsmall": 4,
        "xxxlarge": 48,
        "xxxsmall": 2,
      },
    }
  }
  withBox={true}
>
  <Socials__SocialWrapper>
    <section
      className="c0"
    >
      <Socials__List>
        <ul
          className="c1"
        >
          <ListItem
            key="social-item-facebook"
            theme={
              Object {
                "spacing": Object {
                  "large": 24,
                  "medium": 16,
                  "small": 12,
                  "xlarge": 32,
                  "xsmall": 8,
                  "xxlarge": 40,
                  "xxsmall": 4,
                  "xxxlarge": 48,
                  "xxxsmall": 2,
                },
              }
            }
          >
            <li
              className="c2"
            >
              <SocialType
                key="social-facebook"
                size="48"
                title="facebook title of the purpose of this icon"
                type="facebook"
                url="https://facebook.com"
                withBox={true}
              >
                <SocialType__SocialWrapper
                  href="https://facebook.com"
                >
                  <a
                    className="c3"
                    href="https://facebook.com"
                  >
                    <SocialFacebookIcon
                      size="48"
                      title="facebook title of the purpose of this icon"
                      withBox={true}
                    >
                      <SocialFacebookIcon__StyledSvg
                        aria-labelledby="facebook-title"
                        height="48"
                        role="img"
                        viewBox="0 0 48 48"
                        width="48"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-labelledby="facebook-title"
                          className="c4"
                          height="48"
                          role="img"
                          viewBox="0 0 48 48"
                          width="48"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="facebook-title"
                          >
                            facebook title of the purpose of this icon
                          </title>
                          <g
                            id="Box_Facebook"
                            transform="translate(606 -35)"
                          >
                            <rect
                              data-name="Rectangle 1"
                              fill="#3c5193"
                              height={48}
                              id="Rectangle_1"
                              rx="9"
                              transform="translate(-606 35)"
                              width={48}
                            />
                            <path
                              d="M146.667,100.389h2.285V98.246a5.551,5.551,0,0,1,.737-3.306,4.115,4.115,0,0,1,3.547-1.606,14.874,14.874,0,0,1,4.1.4L156.763,97a7.945,7.945,0,0,0-1.842-.267c-.888,0-1.685.309-1.685,1.165v2.493h3.645l-.254,3.192h-3.391v11.086h-4.284V103.581h-2.285Z"
                              data-name="Path 4"
                              fill="#fff"
                              fillRule="evenodd"
                              id="Path_4"
                              transform="translate(-734 -45)"
                            />
                          </g>
                        </svg>
                      </SocialFacebookIcon__StyledSvg>
                    </SocialFacebookIcon>
                  </a>
                </SocialType__SocialWrapper>
              </SocialType>
            </li>
          </ListItem>
        </ul>
      </Socials__List>
    </section>
  </Socials__SocialWrapper>
</Socials>
`;

exports[`<Socials /> should match snapshots 3`] = `
.c5 {
  display: inline-block;
}

.c4 {
  display: inline-block;
}

.c6 {
  display: inline-block;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  padding: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  list-style: none;
  margin-right: 16px;
}

.c2:last-child {
  margin-right: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<Socials
  items={
    Array [
      Object {
        "name": "facebook",
        "title": "facebook title of the purpose of this icon",
        "url": "https://facebook.com",
      },
      Object {
        "name": "twitter",
        "title": "twitter title of the purpose of this icon",
        "url": "https://twitter.com",
      },
      Object {
        "name": "youtube",
        "title": "youtube title of the purpose of this icon",
        "url": "https://youtube.com",
      },
    ]
  }
  size="24"
  theme={
    Object {
      "spacing": Object {
        "large": 24,
        "medium": 16,
        "small": 12,
        "xlarge": 32,
        "xsmall": 8,
        "xxlarge": 40,
        "xxsmall": 4,
        "xxxlarge": 48,
        "xxxsmall": 2,
      },
    }
  }
  withBox={true}
>
  <Socials__SocialWrapper>
    <section
      className="c0"
    >
      <Socials__List>
        <ul
          className="c1"
        >
          <ListItem
            key="social-item-facebook"
            theme={
              Object {
                "spacing": Object {
                  "large": 24,
                  "medium": 16,
                  "small": 12,
                  "xlarge": 32,
                  "xsmall": 8,
                  "xxlarge": 40,
                  "xxsmall": 4,
                  "xxxlarge": 48,
                  "xxxsmall": 2,
                },
              }
            }
          >
            <li
              className="c2"
            >
              <SocialType
                key="social-facebook"
                size="24"
                title="facebook title of the purpose of this icon"
                type="facebook"
                url="https://facebook.com"
                withBox={true}
              >
                <SocialType__SocialWrapper
                  href="https://facebook.com"
                >
                  <a
                    className="c3"
                    href="https://facebook.com"
                  >
                    <SocialFacebookIcon
                      size="24"
                      title="facebook title of the purpose of this icon"
                      withBox={true}
                    >
                      <SocialFacebookIcon__StyledSvg
                        aria-labelledby="facebook-title"
                        height="24"
                        role="img"
                        viewBox="0 0 48 48"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-labelledby="facebook-title"
                          className="c4"
                          height="24"
                          role="img"
                          viewBox="0 0 48 48"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="facebook-title"
                          >
                            facebook title of the purpose of this icon
                          </title>
                          <g
                            id="Box_Facebook"
                            transform="translate(606 -35)"
                          >
                            <rect
                              data-name="Rectangle 1"
                              fill="#3c5193"
                              height={48}
                              id="Rectangle_1"
                              rx="9"
                              transform="translate(-606 35)"
                              width={48}
                            />
                            <path
                              d="M146.667,100.389h2.285V98.246a5.551,5.551,0,0,1,.737-3.306,4.115,4.115,0,0,1,3.547-1.606,14.874,14.874,0,0,1,4.1.4L156.763,97a7.945,7.945,0,0,0-1.842-.267c-.888,0-1.685.309-1.685,1.165v2.493h3.645l-.254,3.192h-3.391v11.086h-4.284V103.581h-2.285Z"
                              data-name="Path 4"
                              fill="#fff"
                              fillRule="evenodd"
                              id="Path_4"
                              transform="translate(-734 -45)"
                            />
                          </g>
                        </svg>
                      </SocialFacebookIcon__StyledSvg>
                    </SocialFacebookIcon>
                  </a>
                </SocialType__SocialWrapper>
              </SocialType>
            </li>
          </ListItem>
          <ListItem
            key="social-item-twitter"
            theme={
              Object {
                "spacing": Object {
                  "large": 24,
                  "medium": 16,
                  "small": 12,
                  "xlarge": 32,
                  "xsmall": 8,
                  "xxlarge": 40,
                  "xxsmall": 4,
                  "xxxlarge": 48,
                  "xxxsmall": 2,
                },
              }
            }
          >
            <li
              className="c2"
            >
              <SocialType
                key="social-twitter"
                size="24"
                title="twitter title of the purpose of this icon"
                type="twitter"
                url="https://twitter.com"
                withBox={true}
              >
                <SocialType__SocialWrapper
                  href="https://twitter.com"
                >
                  <a
                    className="c3"
                    href="https://twitter.com"
                  >
                    <SocialTwitterIcon
                      size="24"
                      title="twitter title of the purpose of this icon"
                      withBox={true}
                    >
                      <SocialTwitterIcon__StyledSvg
                        aria-labelledby="twitter-title"
                        height="24"
                        role="img"
                        viewBox="0 0 48 48"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-labelledby="twitter-title"
                          className="c5"
                          height="24"
                          role="img"
                          viewBox="0 0 48 48"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="twitter-title"
                          >
                            twitter title of the purpose of this icon
                          </title>
                          <g
                            id="box_twitter"
                            transform="translate(606 -203)"
                          >
                            <rect
                              data-name="twitter-name"
                              fill="#2caae1"
                              height={48}
                              id="twitterID"
                              rx="9"
                              transform="translate(-606 203)"
                              width={48}
                            />
                            <path
                              d="M166.667,262.829a11.931,11.931,0,0,1-3.457.955,6.067,6.067,0,0,0,2.64-3.355,11.977,11.977,0,0,1-3.821,1.478,6.007,6.007,0,0,0-9.872,1.655,6.107,6.107,0,0,0-.383,3.876,17.046,17.046,0,0,1-12.4-6.347,6.1,6.1,0,0,0,1.862,8.092,5.957,5.957,0,0,1-2.731-.755v.077a6.089,6.089,0,0,0,1.367,3.843,6.01,6.01,0,0,0,3.474,2.1,6.017,6.017,0,0,1-2.718.1,6.068,6.068,0,0,0,2.141,3.013,6,6,0,0,0,3.481,1.2,12.071,12.071,0,0,1-8.917,2.51A16.943,16.943,0,0,0,146.559,284c11.07,0,17.123-9.239,17.123-17.252,0-.263,0-.525-.016-.785a12.288,12.288,0,0,0,3-3.134Z"
                              data-name="Path 27"
                              fill="#fff"
                              id="twitterID-path"
                              transform="translate(-734 -45)"
                            />
                          </g>
                        </svg>
                      </SocialTwitterIcon__StyledSvg>
                    </SocialTwitterIcon>
                  </a>
                </SocialType__SocialWrapper>
              </SocialType>
            </li>
          </ListItem>
          <ListItem
            key="social-item-youtube"
            theme={
              Object {
                "spacing": Object {
                  "large": 24,
                  "medium": 16,
                  "small": 12,
                  "xlarge": 32,
                  "xsmall": 8,
                  "xxlarge": 40,
                  "xxsmall": 4,
                  "xxxlarge": 48,
                  "xxxsmall": 2,
                },
              }
            }
          >
            <li
              className="c2"
            >
              <SocialType
                key="social-youtube"
                size="24"
                title="youtube title of the purpose of this icon"
                type="youtube"
                url="https://youtube.com"
                withBox={true}
              >
                <SocialType__SocialWrapper
                  href="https://youtube.com"
                >
                  <a
                    className="c3"
                    href="https://youtube.com"
                  >
                    <SocialYoutubeIcon
                      size="24"
                      title="youtube title of the purpose of this icon"
                      withBox={true}
                    >
                      <SocialYoutubeIcon__StyledSvg
                        aria-labelledby="youtube-title"
                        height="24"
                        role="img"
                        viewBox="0 0 50 50"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-labelledby="youtube-title"
                          className="c6"
                          height="24"
                          role="img"
                          viewBox="0 0 50 50"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="youtube-title"
                          >
                            youtube title of the purpose of this icon
                          </title>
                          <g
                            id="Box_Youtube"
                            transform="translate(350 -35)"
                          >
                            <rect
                              data-name="Rectangle 3"
                              fill="#fff"
                              height={48}
                              id="Rectangle_3"
                              rx="8"
                              stroke="#424242"
                              strokeWidth="2"
                              transform="translate(-349 36)"
                              width={48}
                            />
                            <path
                              d="M422.05,96.9a3.711,3.711,0,0,0-2.6-2.617c-2.283-.617-11.45-.617-11.45-.617s-9.167,0-11.467.617a3.685,3.685,0,0,0-2.6,2.617,39.409,39.409,0,0,0-.6,7.1,38.348,38.348,0,0,0,.617,7.1,3.711,3.711,0,0,0,2.6,2.617c2.283.616,11.45.616,11.45.616s9.167,0,11.467-.616a3.687,3.687,0,0,0,2.6-2.617,39.409,39.409,0,0,0,.6-7.1A38.348,38.348,0,0,0,422.05,96.9Z"
                              data-name="Path 20"
                              fill="#ee3124"
                              id="Path_20"
                              transform="translate(-733 -44)"
                            />
                            <path
                              d="M405,108.367V99.633L412.667,104Z"
                              data-name="Path 21"
                              fill="#fff"
                              id="Path_21"
                              transform="translate(-733 -44)"
                            />
                          </g>
                        </svg>
                      </SocialYoutubeIcon__StyledSvg>
                    </SocialYoutubeIcon>
                  </a>
                </SocialType__SocialWrapper>
              </SocialType>
            </li>
          </ListItem>
        </ul>
      </Socials__List>
    </section>
  </Socials__SocialWrapper>
</Socials>
`;
